app/src/main/java/com/example/cm/ui/adapters/FriendRequestListAdapter.java

```diff
@@ -96,19 +102,94 @@ private void onUndoDelete(FriendRequest request, int position, Request.RequestSt
         FriendRequest request = mRequests.get(position).getValue();
 
-        String user = Objects.requireNonNull(request).getSenderName();
+        String currentUserId = new AuthRepository().getCurrentUser().getUid();
+
+        if (request != null) {
+            if (request.getReceiverId().equals(currentUserId)) {
+                setReceivedRequests(holder, request);
+            }
+            if (request.getSenderId().equals(currentUserId)) {
+                setSentRequests(holder, request);
+            }
+        }
+
+
+    }
+
+    @SuppressLint("SetTextI18n")
+    @RequiresApi(api = Build.VERSION_CODES.N)
+    private void setSentRequests(FriendRequestViewHolder holder, FriendRequest request) {
+        String fullName = getFullName(request.getReceiverId());
+        String userName = getUserName(request.getReceiverId());
+        String date = request.getCreationTimeAgo();
+
+        String requestDescription = "";
+
+        Request.RequestState requestState = request.getState();
+
+        if (requestState == Request.RequestState.REQUEST_PENDING) {
+            requestDescription = "Du wartest auf eine Antwort.";
+        } else if (requestState == Request.RequestState.REQUEST_ACCEPTED) {
+            requestDescription = "Deine Freundschaftsanfrage wurde bestätigt.";
+        } else if (requestState == Request.RequestState.REQUEST_ANSWERED) {
+            requestDescription = "Deine Freundschaftsanfrage wurde beantwortet.";
+        } else if (requestState == Request.RequestState.REQUEST_DECLINED) {
+            requestDescription = "Deine Freundschaftsanfrage wurde abgelehnt.";
+        }
+
+        if (fullName != null) {
+            holder.getTvSender().setText("An: " + fullName);
+        }
+
+        if (userName != null) {
+            holder.getTvSenderUsername().setText(userName);
+        }
+
+        holder.getTvDescription().setText(requestDescription);
+        holder.getTvSentDate().setText(date);
+        holder.getBtnAccept().setVisibility(View.GONE);
+        holder.getBtnDecline().setVisibility(View.GONE);
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.N)
+    private String getFullName(String userId) {
+        MutableLiveData<User> user = users.stream().filter(userData -> Objects.requireNonNull(userData.getValue()).getId().equals(userId)).findAny().orElse(null);
+        if (user != null && user.getValue() != null) {
+            return user.getValue().getFirstName() + " " + user.getValue().getLastName();
+        } else {
+            return null;
+        }
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.N)
+    private String getUserName(String userId) {
+        MutableLiveData<User> user = users.stream().filter(userData -> Objects.requireNonNull(userData.getValue()).getId().equals(userId)).findAny().orElse(null);
+        if (user != null && user.getValue() != null) {
+            return user.getValue().getUsername();
+        } else {
+            return null;
+        }
+    }
+
```

Comment:
Profilbilder mit anzeigen wär noch top, ist aktuell in der xml fürs layout noch auf visibility=gone